### PR TITLE
Job: Implemented retry and retry delay 

### PR DESCRIPTION
**Pull request by Lobarr**

### CODE REVIEW

File: core/block_test.go [hunk 1]
```diff
@@ -25,10 +27,11 @@ func TestNewBlock(t *testing.T) {
 }
 
 func TestVeriyBlock(t *testing.T) {
-	node1 := merkletree.NewNode([]byte("test1asdfasdf job"), &merkletree.MerkleNode{}, &merkletree.MerkleNode{})
-	node2 := merkletree.NewNode([]byte("test2 job asldkj;fasldkjfasd"), &merkletree.MerkleNode{}, &merkletree.MerkleNode{})
-	node3 := merkletree.NewNode([]byte("test3 asdfasl;dfasdjob"), &merkletree.MerkleNode{}, &merkletree.MerkleNode{})
-	node4 := merkletree.NewNode([]byte("tesasdfa;sdasd;laskdjf;alsjflkfj;ast4 job"), &merkletree.MerkleNode{}, &merkletree.MerkleNode{})
+	j := job.NewJob("func test(){return 1+1}; test()")
```
Comment (tktaofik) [inline core/block_test.go]: This is like the name of the job . `"func test(){return 1+1}; test()"` right ??

File: job/engine.go [hunk 2]
```diff
@@ -103,18 +119,46 @@ func (j *Job) Serialize() []byte {
...
-	j.AddExec(exec)
-	return result, err
+	exec.SetTimestamp(time.Now().Unix())
+	exec.SetDuration(time.Duration(time.Now().Sub(start).Nanoseconds()))
+	exec.SetErr(err)
+	exec.SetResult(result)
+	exec.setHash()
+	exec.SetStatus(FINISHED)
+	j.AddExec(*exec)
+	// return &exec
```
Comment (tktaofik) [inline job/engine.go]: don't you want to return an error here ?

Comment (Lobarr) [reply job/engine.go]: The error would be handled by the user. It shouldn't be our problem.

Comment (tktaofik) [reply job/engine.go]: how does the user know an error has occurred ?

File: job/exec.go [hunk 3]
```diff
@@ -44,7 +44,7 @@ func NewJobExec(args []interface{}, retries, priority int, retrydelay time.Durat
 		Retries:       retries,
 		Priority:      priority,
 		Status:        STARTED,
-		RetryDelay:    retrydelay,
+		Backoff:       backoff,
```
Comment (tktaofik) [inline job/exec.go]: I kind of prefer `retryDelay`

File: job/engine.go (hunk 2 above)
Comment (Lobarr) [reply job/engine.go]: Just like an API call, you know about it when it returns

Comment (tktaofik) [reply job/engine.go]: Okay we might still need some form of error handling if things ever fail on our side....its fine in \this case

Comment (Lobarr) [reply job/engine.go]: Yeah for sure

**Pull request merged**